apps/src/lib/util/firehose.js

```diff
@@ -0,0 +1,116 @@
+ *     {
+ *       created_at: new Date(),             // REQUIRED
+ *       environment: 'production',          // REQUIRED
+ *       study: 'underwater basket weaving', // REQUIRED
+ *       study_group: 'control',             // OPTIONAL
+ *       user_id: current_user.id,           // OPTIONAL
+ *       script_id: script.id,               // OPTIONAL
+ *       level_id: level.id,                 // OPTIONAL
+ *       project_id: project.id              // OPTIONAL
+ *       event: 'drowning',                  // REQUIRED
+ *       data_int: 2                         // OPTIONAL
+ *       data_float: 0.31                    // OPTIONAL
+ *       data_string: 'code.org rocks'       // OPTIONAL
+ *       data_json: JSON.stringify(x)        // OPTIONAL
+ *     }
+ *   );
+ * Usage:
+ *   FirehoseClient.putRecordBatch(
+ *     'analysis-events',
+ *     [
+ *       {
+ *         created_at: new Date(),             // REQUIRED
+ *         environment: 'production',          // REQUIRED
+ *         study: 'underwater basket weaving', // REQUIRED
+ *         study_group: 'control',             // OPTIONAL
+ *         user_id: current_user.id,           // OPTIONAL
+ *         script_id: script.id,               // OPTIONAL
+ *         level_id: level.id,                 // OPTIONAL
+ *         project_id: project.id              // OPTIONAL
+ *         event: 'drowning',                  // REQUIRED
+ *         data_int: 2                         // OPTIONAL
+ *         data_float: 0.31                    // OPTIONAL
+ *         data_string: 'code.org rocks'       // OPTIONAL
+ *         data_json: JSON.stringify(x)        // OPTIONAL
+ *       },
+ *       {
+ *         created_at: new Date(),             // REQUIRED
+ *         environment: 'production',          // REQUIRED
+ *         study: 'underwater basket weaving', // REQUIRED
+ *         study_group: 'control',             // OPTIONAL
+ *         user_id: current_user.id,           // OPTIONAL
+ *         script_id: script.id,               // OPTIONAL
+ *         level_id: level.id,                 // OPTIONAL
+ *         project_id: project.id              // OPTIONAL
+ *         event: 'drowning',                  // REQUIRED
+ *         data_int: 2                         // OPTIONAL
+ *         data_float: 0.31                    // OPTIONAL
+ *         data_string: 'code.org rocks'       // OPTIONAL
+ *         data_json: JSON.stringify(x)        // OPTIONAL
+ *       },
+ *     ]
+ *   );
+ */
+// TODO(asher): Add the ability to queue records individually, to be submitted
+// as a batch.
+export default class FirehoseClient {
+  /**
+   * Pushes one data record into the delivery stream.
+   * @param {string} deliveryStreamName The name of the delivery stream.
+   * @param {hash} data The data to push.
+   */
+  putRecord(deliveryStreamName, data) {
+    FIREHOSE.putRecord(
+      {
+        DeliveryStreamName: deliveryStreamName,
+        Record: JSON.stringify(data),
```

Comment:
I'm working on my usage case in a derived branch, and it looks like to get this to work this data needs to be a bit further nested:

      

> {
>         DeliveryStreamName: deliveryStreamName,
>         Record: {
>           Data: JSON.stringify(data)
>         }
>       },